protocol/builder/produce_partition_response_builder.go

```diff
@@ -0,0 +1,57 @@
+package builder
+
+import (
+	kafkaapi "github.com/codecrafters-io/kafka-tester/protocol/api"
+)
+
+type producePartitionResponseBuilder struct {
+	index           int32
+	errorCode       int16
+	baseOffset      int64
+	logAppendTimeMs int64
+	logStartOffset  int64
+	recordErrors    []kafkaapi.ProduceRecordError
+	errorMessage    *string
+}
+
+func NewProducePartitionResponseBuilder() *producePartitionResponseBuilder {
+	return &producePartitionResponseBuilder{
+		index:           -1,
+		errorCode:       0,
+		baseOffset:      0,
+		logAppendTimeMs: -1,
+		logStartOffset:  0,
+		recordErrors:    []kafkaapi.ProduceRecordError{},
+		errorMessage:    nil,
+	}
+}
+
+func (b *producePartitionResponseBuilder) WithIndex(index int32) *producePartitionResponseBuilder {
+	b.index = index
+	return b
+}
+
+func (b *producePartitionResponseBuilder) WithError(errorCode int16) *producePartitionResponseBuilder {
+	b.errorCode = errorCode
+	if errorCode != 0 {
```

Comment:
If we're calling this `WithError`, let's not accept `0` as a value - panic if passed in. The default value is 0 anyway, so for the success case this doesn't need to be called.